### ISSUE TITLE
When running a command Cline UI "locks up" after clicking "Run command"

### ISSUE THREAD

**Issue by dldinternet** [cline/cline#694]
### Which API Provider are you using?

AWS Bedrock

### Which Model are you using?

Claude 3.5 Sonnet v2

### What happened?

This is a basic test scenario where I have Cline run a `go test ` command 
The only thing special is that I run go test WITHOUT `-v` and the test is quiet meaning it does not write to stdout for a long time
Cline cannot be interrupted, new session started or anything. Have to disable the extension or restart VSCode

### Steps to reproduce

1. Run a program that produces no output under Cline control
2. Click "Run Program"
3. Try to interact with Cline

**Comment by saoudrizwan** [comment]
Does this happen with all commands or just `go test`?

**Comment by schbrns** [comment]
Have you tried killing the focused terminal? This gets Cline going again.

**Comment by dldinternet** [comment]
> Does this happen with all commands or just `go test`?

Have not tested widely so not sure. I was very focused on getting this project done :)

**Comment by dldinternet** [comment]
> Have you tried killing the focused terminal? This gets Cline going again.

I have not tried this yet but I will if / when it happens again.

**Comment by mgunnin** [comment]
I'm getting the same issues when running npm/npx commands. Last one happened when it ran "npm run dev". The only way to get it back is to completely restart cursor.

**Comment by sammcj** [comment]
I've had this as well, it Cline sits waiting for the command to finish - but it's actually already finished. The workaround I've found is to click into the terminal and press enter, then cline seems to wake back up again. It's annoying for sure.

**Comment by P4R4DiSi4C** [comment]
Having the same issue here

**Comment by reinierweerts** [comment]
![image](https://github.com/user-attachments/assets/b107d02e-028b-4550-b431-1f7f6a85710d)
Same here, python command gone wrong and no way to resume.

**Comment by DeconstructingAmbiguity** [comment]
Same here.

**Comment by ChristianWeyer** [comment]
For me, this happens waaaay too often... @saoudrizwan

**Comment by reinierweerts** [comment]
"I" created a workaround for my purpose using Claude by creating a 'terminal' MCP server that launches commands (django dev server in my case) and reroutes the server output to a log file I can then tail -f in another terminal. Cline can start and stop the server and log the output. Didn't test with other commands, but at least this works somewhat.

**Comment by tachyon-beep** [comment]
This happens nearly 100% of the time, I have to cancel and restart which often causes VSCode to hang. It's new behaviour in the last 24 hours or so.

**Comment by reinierweerts** [comment]
Can confirm setting fish as default terminal solves the issues.

**Comment by seriouscatsoserious** [comment]
Had some success by killing all terminals by right clicking in them and then killing. After that seems to work for remainder of session.

**Comment by ChristianWeyer** [comment]
Yeah, this is a showstopper, unfortunately @saoudrizwan - thanks for looking into it.

**Comment by ChristianWeyer** [comment]
... I had to switch to Windsurf for now ... 🙈 @saoudrizwan

**Comment by KJ7LNW** [comment]
I've seen this behavior while testing #1089 . In my case, closing the task and re-selecting it breaks out of the spinner, but I'm not sure what the cause is and the bug doen't often present itself.

Can you reliably reproduce the problem?

**Comment by ChristianWeyer** [comment]
> I've seen this behavior while testing #1089 . In my case, closing the "Cline task" and re-selecting from the UI it breaks out of the spinner so I can proceed, but I'm not sure what the cause is and the bug doesn't often present itself.
> 
> Can you reliably reproduce the problem?

This also works for me - but with an explicit "Resume Task"...

**Comment by lishenxydlgzs** [comment]
Does your test modify the terminal content, for example generating progress report and continue to modify the numbers. I had experience running vitest and it turned out that the command is generating a lot of output to update the terminal text. The fix was to use a reporter that will output the result in a single json format (example [here](https://github.com/joeychia/online-course/blob/main/package.json#L17)) that's machine friendly.

**Comment by KJ7LNW** [comment]
> Does your test modify the terminal content

no.  I'm running w/ PR #1089 applied, and haven't seen the problem in a while.  Not sure if that is why or not.

**Comment by empiricompany** [comment]
In my case, this happens only within a WSL2 container context. Hit enter or closing the terminal does not resolve the issue, the only way is to terminate the task. It seems that it cannot read the terminal output.

**Comment by gruckion** [comment]
1. Insure VSCode is up to date.

2. For anyone trying to figure this out. If you're using ZSH rename the file.

`mv ~/.zshrc ~/.zshrc-backup`

Then try again. If it no longer gets stuck then it is something within your ZSH file.

**Comment by KJ7LNW** [comment]
there have been many upstream fixes in vscode, so try an -insiders build

**Comment by natesilva** [comment]
I too found that renaming my `.zshrc` file to `.zshrc-backup` fixed the problem. So I used Cline (of course) to troubleshoot.

In my case the problem was the use of Oh My Zsh. I was able to fix the problem by modifying my `.zshrc` to only load Oh My Zsh if _not_ running within VS Code’s integrated terminal:

```shell
# within my .zshrc file, at the spot where I source `oh-my-zsh.sh`, change to:

if [[ "$TERM_PROGRAM" != "vscode" ]]; then
  source $ZSH/oh-my-zsh.sh
else
  # A simple prompt for VSCode
  PS1='%n@%m %~ %# '
fi
```

**Comment by Jouwert** [comment]
I've ran into this problem as well. A 'stop sign' mouse icon keeps floating over the prompt area after clicking Run Command. In my case it is after trying to run a command changing the folder structure.

**Comment by jeffreyschultz** [comment]
I have found that if I switch the mode from Plan <-> Act then I am able to resume.

**Comment by frostbournesb** [comment]
This should now have been fixed. Can someone say if this is still an issue or may I close this?

**Comment by jeffreyschultz** [comment]
I am no longer having this issue.

**Comment by piiq** [comment]
@frostbournesb 
> Can someone say if this is still an issue or may I close this?

Migrating to Cline from Cursor just now and I have this problem with Cline in VSCode. I'm on a mac and I see this in both cases - me using Cline in a native arm64 environment and when VSCode is attached to a docker container running amd64 architecture. 

I am able to reproduce the following behaviour:
Cline hangs up when trying to run simple commands in the terminal. When I allow it to run a command it creates a new terminal, triggers the command to execute and freezes. My understanding is that there seems to be a problem with how it can get the results to unblock itself and continue (I actually don't know if it should continue, just want it un-stuck)

Un-freezing it in my case is done by creating a new chat and then going back to the chat that produced an error. Once this is done the - there the "Resume task" button is present, but the problem persists in subsequent calls.

The prompts starts with:

```
[TASK RESUMPTION] This task was interrupted just now. It may or may not be complete, so please reassess the task context. Be aware that the project state may have changed since then. The current working directory is now '/workspace'. If the task has not been completed, retry the last step before interruption and proceed with completing the task.

Note: If you previously attempted a tool use that the user did not provide a result for, you should assume the tool use was not successful and assess whether you should retry. If the last tool was a browser_action, the browser has been closed and you must launch a new browser if needed.
```

Here's a screenshot of how it looks like

<img width="629" alt="Image" src="https://github.com/user-attachments/assets/8a7a567a-76e9-41b8-817d-8e6758949398" />

If it's me doing something wrong, any advice would be very much appreciated

**Comment by celestial-vault** [comment]
Thanks for reporting! This has been consolidated into #4356 which provides a comprehensive solution for terminal integration reliability issues across platforms and shell configurations.
